app/repositories/LookupRepository.scala

```diff
@@ -27,34 +27,35 @@ import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
 
 trait LookupRepository {
 
-  def save(nino: String, mtdId: String): Future[Boolean]
+  def save(reference: MtdIdCached): Future[Boolean]
 
-  def getMtdReference(nino: String): Future[Option[MtdIdReference]]
+  def getMtdReference(nino: String): Future[Option[MtdIdCached]]
 
 }
 
 @Singleton
 class LookupRepositoryImpl @Inject() (mongo: MongoComponent)(implicit ec: ExecutionContext)
-    extends PlayMongoRepository[MtdIdReference](
+    extends PlayMongoRepository[MtdIdCached](
       collectionName = "mtdIdLookup",
       mongoComponent = mongo,
-      domainFormat = MtdIdReference.format,
+      domainFormat = MtdIdCached.format,
       indexes = Seq(
         IndexModel(ascending("nino"), IndexOptions().name("mtd-nino").unique(true).background(true))
       ),
       replaceIndexes = false
     )
     with LookupRepository {
 
-  def save(nino: String, mtdId: String): Future[Boolean] =
-    collection.insertOne(MtdIdReference(nino, mtdId)).toFuture().map(result => result.wasAcknowledged())
+  def save(reference: MtdIdCached): Future[Boolean] =
+    Try { collection.insertOne(reference).toFuture().map(result => result.wasAcknowledged()) }.getOrElse(Future.successful(false))
```

Comment:
Will leave for now

Reply:
The wrapping in a `Try` doesn't really give us anything (unless we know the Mongo driver is dodgy and susceptible to throwing exceptions rather than returning failed `Future`s).

Also the call to save in the service ignores its return type so why not implement this as something like the following so that it logs failures (but carries on):
```scala
  def save(reference: MtdIdCached): Future[Boolean] =
    collection
      .insertOne(reference)
      .toFuture()
      .map(_ => true)
      .recover { case e =>
        logger.warn("Error saving reference to cache", e)
        false
      }
```

The (pre-existing) `result.wasAcknowledged()` is also a red herring: is has nothing to do with the success of the insert operation.